keps/sig-storage/2639-secret-configmap-protection/README.md

```diff
@@ -0,0 +1,626 @@
+  - To disable Secret protection, undeploy secret-protection-controller.
+  - To disable ConfigMap protection, undeploy configmap-protection-controller.
+  - If the cluster is downgraded to the version that doesn't support lien, this feature should be disabled, or controllers should be undeployed.
+
+### Version Skew Strategy
+
+- As for dependency on features, this feature depends on lien (InUseProtection feature gate), therefore, if the lien feature is disabled, this feature should also be disabled,
+- As for resources, CSI Volume and CSI Snapshot are involved, changes in the API/CRD of these resources especially for their Secret/ConfigMap fields might cause issue. Howerver, this should be compatibility issue for these API/CRDs.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Other
+  - Describe the mechanism: Deploy the controller for each feature:
+    - secret-protection-controller
+    - configmap-protection-controller
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled). No.
+
+###### Does enabling the feature change any default behavior?
+
+Yes. Secrets/ConfigMaps aren't deleted until the resources using them are deleted.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. By undeploying the controllers.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Secrets/ConfigMaps aren't deleted until the resources using them are deleted, again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, e2e tests for secret-protection-controller and configmap-protection-controller cover scenarios where the controller is deployed and undeployed.
```

Comment:
For in-tree feature gate, there are unit tests for feature enablement/disablement.
Since this KEP is adding external controllers, I'm not aware that there are e2e tests for this.

Reply:
Thank you for point it out. Added a description that feature gate disabled case is covered in unit tests.